### ISSUE TITLE
Can't build NOOBS per instructions in NOOBS readme

### ISSUE THREAD

**Issue by IvanExpert** [raspberrypi/noobs#158]
Using a completely fresh install of Ubuntu Server 13.10 32-bit (including both before and after apt-get upgrade and apt-get dist-upgrade), "sudo ./BUILDME.sh" plows along fine for about an hour, and then dies at:

/home/user1/noobs/buildroot/output/host/usr/bin/moc -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I../../host/usr/arm-buildroot-linux-uclibcgnueabi/sysroot/usr/mkspecs/qws/linux-arm-g++ -I. -I../../host/usr/arm-buildroot-linux-uclibcgnueabi/sysroot/usr/include/QtCore -I../../host/usr/arm-buildroot-linux-uclibcgnueabi/sysroot/usr/include/QtNetwork -I../../host/usr/arm-buildroot-linux-uclibcgnueabi/sysroot/usr/include/QtGui -I../../host/usr/arm-buildroot-linux-uclibcgnueabi/sysroot/usr/include -I. -I. bootselectiondialog.h -o moc_bootselectiondialog.cpp
/home/user1/noobs/buildroot/output/host/usr/bin/rcc -name icons icons.qrc -o qrc_icons.cpp
RCC: Error in 'icons.qrc': Cannot find file 'translation_nl.qm'
make[1]: **\* [qrc_icons.cpp] Error 1
make[1]: **\* Waiting for unfinished jobs....
make[1]: Leaving directory `/home/user1/noobs/buildroot/output/build/recovery-1.0'
make: **\* [/home/user1/noobs/buildroot/output/build/recovery-1.0/.stamp_built] Error 2

This is consistently reproducible for me. I'm using VirtualBox 4.3.6 on a Mac as the Ubuntu host machine. Anything you want me to try doing differently, or can I provide any more detail?

**Comment by procount** [comment]
`sudo apt-get install libqt4-dev` allows lrelease to be built and is stored in buildroot/host/usr/bin. This allows a first build of noobs/recovery. However, lupdate is not built by buildroot, so the build could subsequently fail depending on what modifications are made.
To add lupdate to the build, qt.mak needs updating and projects.pro needs patching.

Here is the diff file for the qt.mak changes:
-----------------------------------------------
diff --git a/buildroot/package/qt/qt.mk b/buildroot/package/qt/qt.mk
index 1e763b5..d2f1715 100644
--- a/buildroot/package/qt/qt.mk
+++ b/buildroot/package/qt/qt.mk
@@ -520,7 +520,7 @@ endef
 # Build the list of libraries and plugins to install to the target
 
 QT_INSTALL_LIBS    += QtCore
-QT_HOST_PROGRAMS   += moc rcc qmake lrelease
+QT_HOST_PROGRAMS   += moc rcc qmake lrelease lupdate
 
 ifeq ($(BR2_PACKAGE_QT_GUI_MODULE),y)
 QT_INSTALL_LIBS    += QtGui
-------------------------------------------------

Here is the new buildroot/package/qt/qt-projects.patch file for projects.pro
-------------------------------------------------
--- qt-everywhere-opensource-src-4.8.1.orig/projects.pro.orig	2013-06-06 22:17:01.000000000 -0700
+++ qt-everywhere-opensource-src-4.8.1.orig/projects.pro	2015-01-20 03:08:12.934200743 -0800
@@ -46,6 +46,7 @@
        contains(QT_BUILD_PARTS, tools):include(doc/doc.pri)
     } else:isEqual(PROJECT, translations) {
        !contains(QT_BUILD_PARTS, tools):!wince*:SUBDIRS += tools/linguist/lrelease
+       SUBDIRS += tools/linguist/lupdate
        SUBDIRS += translations
     } else:isEqual(PROJECT, qmake) {
 #      SUBDIRS += qmake

With these 2 files (and the update to README.MD adding the dependency to libqt4-dev), this issue should be resolved.

**Comment by procount** [comment]
Even the above is not sufficient. I also had to symlink lupdate & lrelease in /usr/bin to the .../host/usr/bin versions, but then it did finally work. Looking forward to the proper fix.

**Comment by wilberforce** [comment]
I'm having the same issue on Arch linux. I have manually patched project.pro and added lupdate to the qt.mk. but still get:

(cd /root/noobs/buildroot/output/build/recovery-1.0 ; /root/noobs/buildroot/output/host/usr/bin/qmake -spec qws/linux-arm-g++)
RCC: Error in 'icons.qrc': Cannot find file 'translation_nl.qm'
RCC: Error in 'icons.qrc': Cannot find file 'translation_de.qm'
RCC: Error in 'icons.qrc': Cannot find file 'translation_pt.qm'
RCC: Error in 'icons.qrc': Cannot find file 'translation_ja.qm'
RCC: Error in 'icons.qrc': Cannot find file 'translation_fr.qm'
RCC: Error in 'icons.qrc': Cannot find file 'translation_hu.qm'
RCC: Error in 'icons.qrc': Cannot find file 'translation_fi.qm'
/usr/bin/make -j5 -C /root/noobs/buildroot/output/build/recovery-1.0 all
make[1]: Entering directory '/root/noobs/buildroot/output/build/recovery-1.0'
/root/noobs/buildroot/output/host/usr/bin/rcc -name icons icons.qrc -o qrc_icons.cpp
RCC: Error in 'icons.qrc': Cannot find file 'translation_nl.qm'
Makefile:288: recipe for target 'qrc_icons.cpp' failed
make[1]: *** [qrc_icons.cpp] Error 1
make[1]: Leaving directory '/root/noobs/buildroot/output/build/recovery-1.0'


Any clues?

**Comment by wilberforce** [comment]
I found a work-around - by editing the translation definitions out of the icons.qrc file I was able to get the build to continue.

**Comment by procount** [comment]
@wilberforce - Normally, a `sudo apt-get  qt4-linguist-tools` should be sufficient to get the lupdate and lrelease tools installed in your distro. Executing recovery/updateqm.sh should update the .qm files. This is called from the BUILDMS.sh script. if this does not work, check you can execute lupdate and lrelease and correct any errors in paths or symlinks. If you have qt5 installed that may cause a conflict. Better to try and get the translations working than to remove them, otherwise you may not end up with a full build.

**Comment by wilberforce** [comment]
Thanks @procount . 

I'm using. Arch linux so use pacman rather than apt-get.

The patch above should allow lupdate to be built as part of the host tools. However with this patch in place the build still fails, so it is not a complete fix and the issue is still present.

**Comment by lurch** [comment]
I'm hoping to properly fix this issue once and for all this week, on all distros - Ubuntu/Arch/etc.

In the meantime as a temporary workaround, while investigating #244 recently I found that in order to successfully build on ArchLinux I had to `pacman -S qt4` and then [edit](https://github.com/raspberrypi/noobs/issues/244#issuecomment-77255953) `recovery/updateqm.sh` to append `-qt4` to the `lupdate

**Comment by wilberforce** [comment]
@lunch - thanks. I had done the pacman -s qt4

I'll try that tomorrow.

**Comment by wilberforce** [comment]
The issue on arch seems to be around the path:
[root@archlinux-2015-03-09 recovery-1.0]# more updateqm.sh
#!/bin/sh

lupdate -no-obsolete recovery.pro
lrelease recovery.pro
[root@archlinux-2015-03-09 recovery-1.0]# which lupdate
which: no lupdate in (/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
[root@archlinux-2015-03-09 recovery-1.0]#


When I run as:
/usr/lib/qt4/bin/lupdate -no-obsolete recovery.pro
/usr/lib/qt4/bin/lrelease recovery.pro

The translation files update.

**Comment by kuba-moo** [comment]
The same problem on Fedora. We should just add an explicit `lrelease` step in `buildroot/package/recovery/recovery.mk`, no?
```
(cd $(@D) ; $(HOST_DIR)/usr/bin/lrelease recovery.pro; $(QT_QMAKE))
```

**Comment by lurch** [comment]
@kuba-moo Something [like that](https://github.com/raspberrypi/noobs/issues/158#issuecomment-31316588) is the plan, yes. I just need to find the time to implement it, and then test it on multiple different distros... unfortunately there always seems to be other more important stuff to work on :-/

Apologies to all for me taking so long to get round to fixing this bug, but obviously to test this fully requires doing from-scratch builds of NOOBS on "clean" distro installs (with no QT4 tools already installed), and clean builds of NOOBS take rather a while to compile...

**Comment by procount** [comment]
Even if you install qt4-linguist-tools, if you have qt5 installed, lupdate may not be found because it will be looking for the qt5 version.
I found an easy fix! Just export QT_SELECT=4 and then lupdate from qt4 will be executed instead of the non-existent version from qt5.
(I found that /usr/bin/lupdate was symlinked to qt_chooser. Setting QT_SELECT=4 is the same as adding -qt4 and forces the qt4 version of all tools to be used)
